pkg/registries/registries.go

```diff
@@ -104,18 +110,31 @@ func mergedMirrorSets(icspRules []*apioperatorsv1alpha1.ImageContentSourcePolicy
 func EditRegistriesConfig(config *sysregistriesv2.V2RegistriesConf, insecureRegistries, blockedRegistries []string, icspRules []*apioperatorsv1alpha1.ImageContentSourcePolicy) error {
 	// getRegistryEntry returns a pointer to a modifiable Registry object corresponding to scope,
 	// creating it if necessary.
-	// NOTE: We never generate entries with Prefix != Location, so everything in updateRegistriesConfig
-	// only checks Location.
+	// Note: Prefix can be a wildcard entry in the format "*.example.registry.com" so we check the value of
+	// Prefix for a wildcard entry. If Prefix doesn't have a wildcard entry, we check Location for regular entries.
 	// NOTE: The pointer is valid only until the next getRegistryEntry call.
 	getRegistryEntry := func(scope string) *sysregistriesv2.Registry {
 		for i := range config.Registries {
-			if config.Registries[i].Location == scope {
+			reg := config.Registries[i].Location
+			if isValidWildcard(config.Registries[i].Prefix) {
+				reg = config.Registries[i].Prefix
+			}
+			if reg == scope {
 				return &config.Registries[i]
 			}
 		}
-		config.Registries = append(config.Registries, sysregistriesv2.Registry{
-			Endpoint: sysregistriesv2.Endpoint{Location: scope},
-		})
+		// If scope is a valid wildcard entry, add it to the registry Prefix
+		if isValidWildcard(scope) {
+			config.Registries = append(config.Registries, sysregistriesv2.Registry{
+				Prefix: scope,
+			})
+			// Only add scope to the endpoint Location if it is not a wildcard entry and it
+			// doesn't containe the wildcard * in the wrong location.
+		} else if !strings.Contains(scope, "*") {
```

Comment:
Why is it acceptable to _silently_ not do what the caller was expecting?
The `return` below would just silently return a completely wrong registry entry.

Reply:
Fixed.
We check if the scope has prefix `*.`, if it does we add scope to registry.Prefix, otherwise we add it to registry.Endpoint.Location.